modules/react/_examples/stories/examples/SidePanelWithNavigation.tsx

```diff
@@ -0,0 +1,238 @@
+import * as React from 'react';
+import {styled, StyledType} from '@workday/canvas-kit-react/common';
+import {colors, space, gradients, type} from '@workday/canvas-kit-react/tokens';
+import {SidePanel, useSidePanel} from '@workday/canvas-kit-preview-react/side-panel';
+import {Flex, Box} from '@workday/canvas-kit-react/layout';
+import {BodyText, Subtext} from '@workday/canvas-kit-react/text';
+import {Expandable} from '@workday/canvas-kit-labs-react/expandable';
+import {SystemIcon} from '@workday/canvas-kit-react/icon';
+import {rocketIcon, notificationsIcon} from '@workday/canvas-system-icons-web';
+
+const StyledPanel = styled(SidePanel)({
+  backgroundColor: colors.soap200,
+  zIndex: 1,
+});
+
+const StyledExpandable = styled(Expandable.Target)({
+  borderRadius: 0,
+  ':hover': {
+    backgroundColor: colors.soap500,
+  },
+});
+
+const StyledLink = styled(Subtext.as('a'))<StyledType>({
+  color: colors.blackPepper300,
+  ':hover': {
+    backgroundColor: colors.soap500,
+  },
+});
+
+const forDesigners = [
+  {value: 'Understanding Canvas', linkValue: '#UnderstandingCanvas'},
+  {value: 'Customizing Canvas', linkValue: '#CustomizingCanvas'},
+  {value: 'Contributing to Canvas', linkValue: '#ContributingtoCanvas'},
+  {value: 'Resources', linkValue: '#Resources'},
+];
+
+const forDevelopers = [
+  {value: 'Using Canvas', linkValue: '#UsingCanvas'},
+  {value: 'Contributing', linkValue: '#Contributing'},
+  {value: 'Resources', linkValue: '#Resources'},
+];
+
+const gettingStarted = [
+  'Extended Systems',
+  'Canvas Glossary',
+  'Frequently Asked Questions',
+  'Canvas Support',
+];
+
+export const WithNavigation = () => {
+  const {labelProps} = useSidePanel();
+
+  return (
+    <Flex height={800} position="relative" backgroundColor="frenchVanilla100">
+      <StyledPanel touched={true} variant="alternate">
+        <Flex flexDirection="column" rowGap="zero" paddingTop="m">
+          <Flex.Item>
+            <Expandable padding="zero">
+              <StyledExpandable paddingY="s" paddingX="m" textAlign="center">
+                <SystemIcon icon={rocketIcon} marginX="xxs" />
+                <Expandable.Title padding="zero">
+                  <BodyText
+                    as="h4"
```

Comment:
We won't want h4 elements in a left nav.

Reply:
Hey @bstanton678! What's the reasoning behind excluding `h4` elements from a left nav? I ask since the Canvas Site side panel navigation (which the example in this PR is modeled after) includes `h4` elements based on recommendations from A11y (albeit in a difference place compared to example):

---

**Example in this PR:**

<img width="1477" alt="Screenshot 2023-03-31 at 11 42 35 AM" src="https://user-images.githubusercontent.com/146020/229203837-f206de65-a3f4-4373-acfb-4ffdaaea9f56.png">

---

**Canvas Site:**

<img width="2037" alt="Screenshot 2023-03-31 at 11 44 10 AM" src="https://user-images.githubusercontent.com/146020/229205042-b48d7af6-a6d5-45b5-868e-0474c7bcdf92.png">

I understand it looks like we're skipping heading levels with the `h3` for `Getting Started` given that there's no `h2` above it (like the `ELEMENTS` `h2` you see a little further down the nav).

---

**Recommendations from A11y:**

<img width="886" alt="image" src="https://user-images.githubusercontent.com/146020/229205907-26fb9f65-5590-4c32-94b1-9e77460b4bb6.png">

Reply:
Ah, thanks for sharing this context @jamesfan , I'd like to sync up on these written requirements and find a way to align. 

From my PoV, a navigation panel doesn't actually have any "content" in it, the panel is simply a structured collection of buttons or links that let users navigate to the content on the site. That's why I didn't feel heading markup was appropriate in this component. 

We can convey hierarchal structure by using nested unordered lists, which I asked for further down this PR. 

From a usability PoV, Headings are very valuable navigational tools, and if I want to use a screen reader's heading navigation capability to get to a page's main content, it is very painful that I need to keep jumping through this navigation panel every time. Bold font weight !== to headings in every case. Only when it is used as a means of describing a block of content.

Reply:
Agreed that a navigation panel doesn't really qualify as "content" -- if headings should only be used in conjunction with content, then I see the issue here.

Reply:
Removed mention of h4